presto-orc/src/main/java/com/facebook/presto/orc/checkpoint/Checkpoints.java

```diff
@@ -83,9 +85,9 @@ private Checkpoints()
             int sequence = entry.getKey().getSequence();
             List<Integer> positionsList = entry.getValue().get(rowGroupId).getPositions();
 
-            ColumnEncodingKind columnEncoding = columnEncodings.get(column).getColumnEncodingKind();
+            ColumnEncodingKind columnEncoding = columnEncodings.get(column).getColumnEncoding(sequence).getColumnEncodingKind();
             OrcTypeKind columnType = columnTypes.get(column).getOrcTypeKind();
-            Set<StreamKind> availableStreams = streamKinds.get(column);
+            Set<StreamKind> availableStreams = streamKinds.get(new ColumnAndSequence(entry.getKey()));
```

Comment:
nit: I would have a constructor over column and sequence, so we can write 
```java
    Set<StreamKind> availableStreams = streamKinds.get(new ColumnAndSequence(column, sequence));
```

Reply:
I originally had that, my one concern was that column and sequence are both ints, so it'd be easy to swap them accidentally.  By moving the logic into the constructor, we have
this.column = stream.getColumn();
this.sequence = stream.getSequence();
and it's really obvious what maps to what

Granted given how contained this code is, I may be overthinking this.